python -m test.framework.suite failed

$ type module
module is a function
module () 
{ 
    eval $($LMOD_CMD bash "$@") && eval $(${LMOD_SETTARG_CMD:-:} -s sh)
}

$ type -f module
bash: type: module: not found

$ module --version

Modules based on Lua: Version 7.7  2017-09-19 11:05 -05:00
    by Robert McLay mclay@tacc.utexas.edu

$ module av EasyBuild

--------------------------------------------------------------------------------------------------------------------------- /home/yk336/.local/easybuild/modules/all ---------------------------------------------------------------------------------------------------------------------------
   EasyBuild/3.4.0 (L)

  Where:
   L:  Module is loaded

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".

$ which -a eb
/home/xxx/.local/easybuild/software/EasyBuild/3.4.0/bin/eb

$ eb --version
This is EasyBuild 3.4.0 (framework: 3.4.0, easyblocks: 3.4.0) on host zzz.


```
INFO: This is (based on) vsc.install.shared_setup 0.10.26

Running tests...
----------------------------------------------------------------------
......Deprecated functionality, will no longer work in v10000000000000: almost kaput; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
Deprecated functionality, will no longer work in v10000000000000: almost kaput; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
Deprecated functionality, will no longer work in v10000000000000: almost kaput; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
Deprecated functionality, will no longer work in v10000000000000: almost kaput; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
Deprecated functionality, will no longer work in v10000000000000: almost kaput; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
....................................................................== backup of existing module file stored at /tmp/eb-JCISF6/eb-8vhkgT/eb-n7MigL/tmpE1tNyR/modules/all/toy/0.0.lua.bak_20170921150824
== comparing module file with backup /tmp/eb-JCISF6/eb-8vhkgT/eb-n7MigL/tmpE1tNyR/modules/all/toy/0.0.lua.bak_20170921150824; no differences found
.............F(skipping HgRepository test)
..(skipping SvnRepository test)
................................Skipping test_dep_graph, since pygraph is not available
....Skipping test_dump_autopep8, since autopep8 is not available
..............Deprecated functionality, will no longer work in v4.0: Named argument 'default_fallback' for get_easyblock_class is deprecated, use 'error_on_missing_easyblock' instead; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
.........................................................................................................Skipping timeout test in test_download_file (working offline)
.......F...................................................== installing extension ext1  (1/1)...
.................F== installing extension ext1  (1/1)...
......................................................
WARNING: Found one or more non-allowed loaded (EasyBuild-generated) modules in current environment:
* toy/0.0-external-deps

This is not recommended since it may affect the installation procedure(s) performed by EasyBuild.

To make EasyBuild allow particular loaded modules, use the --allow-loaded-modules configuration option.
To specify action to take when loaded modules are detected, use --detect-loaded-modules={error,ignore,purge,unload,warn}.

See http://easybuild.readthedocs.io/en/latest/Detecting_loaded_modules.html for more information.


WARNING: Found one or more non-allowed loaded (EasyBuild-generated) modules in current environment:
* toy/0.0-external-deps

This is not recommended since it may affect the installation procedure(s) performed by EasyBuild.

To make EasyBuild allow particular loaded modules, use the --allow-loaded-modules configuration option.
To specify action to take when loaded modules are detected, use --detect-loaded-modules={error,ignore,purge,unload,warn}.

See http://easybuild.readthedocs.io/en/latest/Detecting_loaded_modules.html for more information.


WARNING: Found one or more non-allowed loaded (EasyBuild-generated) modules in current environment:
* toy/0.0-external-deps

This is not recommended since it may affect the installation procedure(s) performed by EasyBuild.

To make EasyBuild allow particular loaded modules, use the --allow-loaded-modules configuration option.
To specify action to take when loaded modules are detected, use --detect-loaded-modules={error,ignore,purge,unload,warn}.

See http://easybuild.readthedocs.io/en/latest/Detecting_loaded_modules.html for more information.

.....== backup of existing module file stored at /tmp/eb-JCISF6/eb-8vhkgT/eb-z2XPBd/tmpAFfllQ/modules/all/toy/0.0.lua.bak_20170921151316
== comparing module file with backup /tmp/eb-JCISF6/eb-8vhkgT/eb-z2XPBd/tmpAFfllQ/modules/all/toy/0.0.lua.bak_20170921151316; no differences found
...............== backup of existing module file stored at /tmp/eb-JCISF6/eb-8vhkgT/eb-YJSnha/tmpENujR5/modules/all/toy/0.0.lua.bak_20170921151354
== comparing module file with backup /tmp/eb-JCISF6/eb-8vhkgT/eb-YJSnha/tmpENujR5/modules/all/toy/0.0.lua.bak_20170921151354; diff is:
--- /tmp/eb-JCISF6/eb-8vhkgT/eb-YJSnha/tmpENujR5/modules/all/toy/0.0.lua.bak_20170921151354
+++ /tmp/eb-JCISF6/eb-8vhkgT/eb-YJSnha/tmpENujR5/modules/all/toy/0.0.lua
@@ -17,7 +17,7 @@
 
 conflict("toy")
 if not userInGroup("xxx") then
-    LmodError("You are not part of 'xxx' group of users that have access to this software; Please consult with user support how to become a member of this group")
+    LmodError("Hey, you're not in the 'xxx' group!")
 end
 
 setenv("EBROOTTOY", root)

..........................................................................................................................
======================================================================
ERROR [1.020s]: test_gitrepo (test.framework.repository.RepositoryTest)
Test using GitRepository.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/xxx/.local/easybuild/software/EasyBuild/3.4.0/lib/python2.7/site-packages/easybuild_framework-3.4.0-py2.7.egg/test/framework/repository.py", line 110, in test_gitrepo
    self.assertTrue(log_regex.search(logmsg), "Pattern '%s' found in %s" % (log_regex.pattern, logmsg))
AssertionError: None is not true : Pattern 'toy/0.0 with EasyBuild v3.4.0 @ .* \(time: .*, user: .*\)' found in commit 8456f867b03aa001fd5a6fe5a0c4300145c065dc
Author: Jens Timmerman <jens.timmerman@ugent.be>
Date:   Wed Oct 24 16:01:05 2012 +0200

    added some more files

======================================================================
ERROR [0.277s]: test_find_eb_script (test.framework.filetools.FileToolsTest)
Test find_eb_script function.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/xxx/.local/easybuild/software/EasyBuild/3.4.0/lib/python2.7/site-packages/easybuild_framework-3.4.0-py2.7.egg/test/framework/filetools.py", line 1356, in test_find_eb_script
    self.assertErrorRegex(EasyBuildError, "Script 'no_such_script' not found", ft.find_eb_script, 'no_such_script')
  File "/home/xxx/.local/easybuild/software/EasyBuild/3.4.0/lib/python2.7/site-packages/vsc_install-0.10.26-py2.7.egg/vsc/install/testing.py", line 126, in assertErrorRegex
    self.assertTrue(regex.search(msg), "Pattern '%s' is found in '%s'" % (regex.pattern, msg))
AssertionError: None is not true : Pattern 'Script 'no_such_script' not found' is found in 'Resolving path None failed: 'NoneType' object has no attribute 'startswith''

======================================================================
ERROR [0.360s]: test_prepare_step (test.framework.easyblock.EasyBlockTest)
Test prepare step (setting up build environment).
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/xxx/.local/easybuild/software/EasyBuild/3.4.0/lib/python2.7/site-packages/easybuild_framework-3.4.0-py2.7.egg/test/framework/easyblock.py", line 1148, in test_prepare_step
    self.assertEqual(self.modtool.list(), [])
  File "/home/xxx/.local/easybuild/software/EasyBuild/3.4.0/lib/python2.7/site-packages/vsc_install-0.10.26-py2.7.egg/vsc/install/testing.py", line 79, in assertEqual
    raise AssertionError("%s:\nDIFF%s:\n%s" % (msg, limit, ''.join(diff[:self.ASSERT_MAX_DIFF])))
AssertionError: Lists differ: [{'default': None, 'mod_name':... != []

First list contains 1 additional elements.
First extra element 0:
{'default': None, 'mod_name': 'EasyBuild/fake'}

- [{'default': None, 'mod_name': 'EasyBuild/fake'}]
+ []:
DIFF:
- [{'default': None, 'mod_name': 'EasyBuild/fake'}]

----------------------------------------------------------------------
Ran 518 tests in 567.475s

FAILED (errors=3)

Generating XML reports...
ERROR: Not all tests were successful.
Log available at /tmp/eb-JCISF6/eb-8vhkgT/easybuild-tests-S238PZ.log , XML output of tests available in test-reports directory
```

--- a/easybuild/tools/repository/gitrepo.py
+++ b/easybuild/tools/repository/gitrepo.py
@@ -103,7 +103,7 @@ class GitRepository(FileRepository):
             client = git.Git(self.wc)
             client.clone(self.repo)
             reponame = os.listdir(self.wc)[0]
-            self.log.debug("rep name is %s" % reponame)
+            self.log.debug("Repo name is %s", reponame)
         except (git.GitCommandError, OSError) as err:
             # it might already have existed
             self.log.warning("Git local repo initialization failed, it might already exist: %s", err)
@@ -111,7 +111,7 @@ class GitRepository(FileRepository):
         # local repo should now exist, let's connect to it again
         try:
             self.wc = os.path.join(self.wc, reponame)
-            self.log.debug("connectiong to git repo in %s" % self.wc)
+            self.log.debug("Connecting to git repo in %s", self.wc)
             self.client = git.Git(self.wc)
         except (git.GitCommandError, OSError) as err:
             raise EasyBuildError("Could not create a local git repo in wc %s: %s", self.wc, err)
@@ -119,7 +119,7 @@ class GitRepository(FileRepository):
         # try to get the remote data in the local repo
         try:
             res = self.client.pull()
-            self.log.debug("pulled succesfully to %s in %s" % (res, self.wc))
+            self.log.debug("Pulled succesfully to %s in %s", res, self.wc)
         except (git.GitCommandError, OSError) as err:
             raise EasyBuildError("pull in working copy %s went wrong: %s", self.wc, err)
 

--- a/test/framework/repository.py
+++ b/test/framework/repository.py
@@ -33,7 +33,7 @@ import shutil
 import sys
 import tempfile
 from test.framework.utilities import EnhancedTestCase, TestLoaderFiltered
-from unittest import TextTestRunner
+from unittest import TextTestRunner, mock
 
 from easybuild.framework.easyconfig.parser import EasyConfigParser
 from easybuild.tools.build_log import EasyBuildError
@@ -102,7 +102,9 @@ class RepositoryTest(EnhancedTestCase):
             repo.init()
             toy_ec_file = os.path.join(os.path.dirname(__file__), 'easyconfigs', 'test_ecs', 't', 'toy', 'toy-0.0.eb')
             repo.add_easyconfig(toy_ec_file, 'test', '1.0', {}, None)
-            repo.commit("toy/0.0")
+            with mock.patch.dict(os.environ, {'GIT_AUTHOR_NAME': 'test', 'GIT_AUTHOR_EMAIL': 'test@test.org',
+                                              'GIT_COMMITTER_NAME': 'test', 'GIT_COMMITTER_EMAIL': 'test@test.org'}):
+                repo.commit("toy/0.0")
 
             log_regex = re.compile(r"toy/0.0 with EasyBuild v%s @ .* \(time: .*, user: .*\)" % VERSION, re.M)
             logmsg = repo.client.log('HEAD^!')

